API Proposal: Updated APIs for channel member refactor

jwilander opened the pull request:
Proposal for API changes in mattermost/platform#4106

enahum commented on source/channels.yaml:
```diff
@@ -142,3 +142,58 @@
+          type: string
+      responses:
+        '200':
+          description: The channel stats
```
should this be `channel member` instead of channel stats?

jwilander replied:
Oops, that's right

enahum commented on source/definitions.yaml:
```diff
@@ -105,6 +105,14 @@ definitions:
+  ChannelStats:
+    type: object
+    properties:
+      id:
```
I prefer to be more consistent here maybe as with ChannelMember where `id` is actually `channel_id`?

jwilander replied:
That was from the legacy stuff, but yes that's a good idea. I'll update

enahum commented on source/users.yaml:
```diff
@@ -114,8 +115,101 @@
+          type: string
+        - name: channel_id
+          in: path
+          description: The channel ID
```
`The channel ID the users are not part of` ? to be more explicit

jwilander replied:
Agreed, will update

jwilander commented:
@enahum Changes made, take a look

jwilander commented:
I made some updates to the new APIs:
* Changed `/users/profiles_by_ids` to `/users/ids` to be more consistent with REST and to be consistent with future similar APIs.
* Changed `/teams/{team_id}/channels/{channel_id}/member/{user_id}` to '/teams/{team_id}/channels/{channel_id}/members/{user_id}' to keep consistent resource name for `members` APIs

enahum commented:
+1
